src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -546,6 +528,47 @@ export default class Pilot {
         return [true, readback];
     }
 
+    /**
+     * Cross a fix at a certain altitude
+     *
+     * @for Pilot
+     * @method cross
+     * @param aircraft {AircraftModel}
+     * @param fixName {string} name of the fix
+     * @param altitude {number} the altitude
+     * @return {array}                [success of operation, readback]
+     */
+    cross(aircraft, fixName, altitude) {
+        if (!NavigationLibrary.hasFixName(fixName)) {
+            return [false, `unknown fix '${fixName}'`];
+        }
+
+        if (!this._fms.hasWaypointName(fixName)) {
+            return [false, `fix '${fixName}' is not on our route`];
+        }
+
+        const response = aircraft.validateNextAltitude(altitude);
```

Comment:
same principle here, if `.validateNextAltitude()` always returns an array this if is much simpler.

Reply:
done